---
title: "Applying linear models with GWAS"
format: html
editor: source
---

```{r}
#| label: load-packages
#| include: false

library(tidyverse)
library('DT')
library(LaCroixColoR)
palette(lacroix_palette("Mango", type = "discrete"))
library(stringr)
library(car)
```

## What are we doing here?

We've spent all semester learning about general linear models, and this lesson will spend some time on one specific application: genome-wide association studies. Even if your research is totally unrelated, I hope that seeing how what we've learnied about general linear models applies to a specific research topic will be useful for relating what we've learned to your own research.

## Linking genes to traits

The basic idea for most (all?) genetic mapping is trying to predict phenotypes from genotypes. Over time, geneticists have used a number of different strategies to make these predictions, from pedigree-based mapping to QTL mapping to genome-wide association studies, which we'll discuss today. 

### Single locus association tests
For a single locus, we can write out a basic model as follows:

$$ Y_i = b_0 + b_1 \times X_i + e_i $$

where $Y_i$ is the trait value for the $i^{th}$ individual, $b_0$ is an intercept, $b_1$ is the effect of our locus, $e_i$ is a residual, and $X_i$ is the genotype of the $i^{th}$ individual. In diploids, and for a biallelic locus, we can set one allele to 0 and define $X_i$ as the number of alternate alleles the individual carries. So if the two alleles are **A** and **T**, genotype **AA** would be 0, **AT** would be 1, and **TT** would be 2. 

![A reproduction of Figure 2D from Enbody et al. 2023 showing the effect of the S and L alleles at the G01 locus on beak size in Darwin's finches. DOI: 10.1126/science.adf6218 ](figs/birdbeaks.png){#fig-birdbeaks fig-alt="A plot with genotype on the X axis and beak size (PC1) on the Y axis. There are three categories of genotypes: SS, SL, and LL, each with many points and the points are overlaid with black box and whisker plots. While there is overlap between categories it is clear that LL genotypes have larger PC1 values than SL genotypes, which have larger values than SS genotypes" fig-align="center" width=50%}

An example is shown in @fig-birdbeaks, of a locus in Darwin's finches (*Geospiza fortis*) that affects beak size. The two alleles in this locus are tagged **S** (for small) and **L** (for large). This paper conducted an ANOVA to estimate effect sizes instead of a linear model, but recall that these techniques are closely related. Ultimately, if the genotype at a locus does not affect phenotype, we expect that the regression line from a linear model will be flat, so $b_1 = 0$, and in an ANOVA, there would be the same amount of variation between groups as within groups.


### Scaling up to genome-wide studies

A genome-wide association study (GWAS) does the analysis that was just described, but at every single-nucleotide polymorphism (SNP) in the genome. So hundreds of thousands or millions of little general linear models looking for a relationship between genotype and phenotype. That's it! 

Looking at hundreds of thousands of scatter plots would be impossible, so GWAS studies tend to report their results in Manhattan plots that show the genome location on the x axis and the negative log of the p value from the statistical test on the y axis. Larger values indicate higher significance. Usually a significance line is plotted on these as well (we'll discuss how this is chosen in a bit).

![A reproduction of Figure 2B from Enbody et al. 2023 for a Manhattan plot summarizing the results of a GWAS for  beak size in Darwin's finches. You can see the G01 locus that was shown in @fig-birdbeaks in chromosome 1 on the left side of the plot DOI: 10.1126/science.adf6218 ](figs/birdbeakmanhattanplot.png){#fig-birdmanhattan fig-alt="A manhattan plot for beak size with multiple peaks" fig-align="center" width=70%}

## Complications to the simple model

I've described GWAS as a relatively simple set of lots of general linear models. But, there are a some things about them that make them more complicated. 

### Multiple testing

GWAS involve doing lots of tests, and so, as we learned in @sec-multtesting, we expect to get lots of p values below an $\alpha$ of 0.05, even if there is no association between any genotypes and phenotype. GWAS studies tend to use either a Bonferroni correction or an FDR analysis (or sometimes both!).

### Population structure
Recall that one of the assumptions of the general linear model is that all of the data has been independently sampled. But, a GWAS sample can violate this assumption. 

We know that all living organisms are related to each other, and that some are more closely related to each other than others. This variation in genetic relatedness is called **population structure**. For example, you are more closely related to your siblings and cousins than you are to me (unless there's something funny going on) so your genome will be more similar to your sibilings and cousins than it is to mine. Similarly, when Enbody et al. sampled finches, they found that there was variation in relatedness within *G. fortis* (the species they did the GWAS on), which you can see in @fig-finchancestry.

![A reproduction of Figure 1A&B from Enbody et al. 2023 showing relatedness between different birds sampled in their study DOI: 10.1126/science.adf6218 ](figs/finchancestry.png){#fig-finchancestry fig-alt="A panel showing ink drawings of four finch species (G. scandens, G. magnirostris, G. fortis, and G. fuliginosa), a tree showing relatedness between all samples. The lartest clade is G. fortis. Below the tree is a structure plot of ancestry showing that G. fortris individuals have some introgression from other species" fig-align="center" width=70%}

Imagine that two distantly related individuals have very different beak sizes because they differ at the G01 allele (plotted in @fig-birdbeaks). However, they will also differ at lots of other loci because they are distantly related, so a GWAS could find many loci across the genome associated with the trait. 

To avoid this problem, many GWAS studies that use samples from across a range will include a "control" for population structure. The most common approach is to use a matrix describing genetic similarity (the 'kinship matrix') between samples as a **random effect**, turning the general linear model into a **mixed model**. These are concepts that you will learn more about in 831, but we will preview them now. The idea of random effects is that they can capture variance in the predictions made by a linear model, but are for factors where we are not interested in predicting specific values for specific individuals. 

[**Mixed models** combine random and fixed effects.]{.aside}

[A **Kinship matrix** or **relatedness matrix** is a matrix describing the genetic similarity of all individuals in a sample. It is technically a 'variance-covariance' matrix.]{.aside}

@fig-ricegwas shows examples of a GWAS conducted for plant height in rice with, and without, a kinship matrix included as a random effect. Without a kinship matrix, there are many loci across the genome that are associated with height but adding a kinship matrix reduces the number of significant associations.

![A reproduction of Figure 3D from Zhao et al. 2011 showing a GWAS done with a standard general linear model (bottom) and a mixed model including a relatedness matrix as a random effect (bottom) https://doi.org/10.1038/ncomms1467 ](figs/ricegwas.png){#fig-ricegwas fig-alt="Two manhattan plots. They look relatively similar but the top one has negative log p values that max out around 6, while the bottom one has lots of negative log p values in the 10-30 range" fig-align="center" width=70%}

## Conducting a GWAS in R

You can conduct a GWAS in using [GAPIT](https://github.com/jiabowang/GAPIT). Check out their website for information on installation. I'd suggest installing it with the following approach:

```{r, eval=F}
install.packages('devtools')
library('devtools')
devtools::install_github("jiabowang/GAPIT",force=TRUE)

```

Here is an example of GAPIT using its tuturial dataset of maize data.
```{r, eval=F}
library(GAPIT)

myY <- read.table("data/mdp_traits.txt", head = TRUE)
myG <- read.table("data/mdp_genotype_test.hmp.txt", head = FALSE)
#Step 2: Run GAPIT
myGAPIT <- GAPIT(
Y=myY[,1:2],
G=myG,
file.output = FALSE
)
```

