---
title: "lab5.2-gwas"
output: word_document
date: "2025-11-19"
---

# Lab 5 (part 2) GWAS.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
```


Actually running a GWAS can be a huge pain because the data files are large and often use irritating formats (Emily spent a few hours wrangling files before abandoning the idea of you all running your own GWASs on your laptops). Instead, you'll be using the results of a GWAS that Emily has already run for you (you can see the code here: https://github.com/bobvanburen/PLB812_F25_GWAS/blob/main/gwas-lab-2025.ipynb)

The species: *Arabidopsis thaliana*, a cute little herbaceous weed found all over Europe, often in disturbed habitats.

The trait: Flowering time, measured in a growth chamber. This trait is really important shaping plant fitness.

First, here are results for a GWAS with a mixed model. The code below loads in the results as a data table.
```{r}
mixedGwas <- read.table('~/Downloads/fl-mixed.assoc.txt', header=T)

mixedGwas$fdr <- p.adjust(mixedGwas$p_lrt, method="fdr")# calculate fdr

dplyr::filter(mixedGwas, fdr<0.01)

```


GWAS without a mixed model:

```{r}
refGwas <- read.table('~/Downloads/fl.assoc.txt', header=T)

refGwas$fdr <- p.adjust(mixedGwas$p_lrt, method="fdr")# calculate fdr

plot(refGwas$p_lrt, mixedGwas$p_lrt)
```


Look at a random locus and make a plot with genotype on the x axis and trait on the y axis

Look at a hit from the second gwas and make a plot with genotype on the x axis and trait on the y axis. Compare the p values and effect sizes in the two GWAS results tables with what you get from a simple linear model with lm()

```{r}
#install.packages('vcfR')
library(vcfR) #library to read in vcf

# read in polymorphism data about this snp
mysnp <- read.vcfR('data/Chr5_18590501.vcf')

# this is a function to calculate genotype numeric scores from the vcf
sum_genotype <- function(geno_col) {
  sapply(geno_col, function(gt) {
    if (is.na(gt) || gt == "./.") return(NA)  # Handle missing
    alleles <- unlist(strsplit(gt, "/"))      # Split by "/"
    sum(as.numeric(alleles))                  # Sum the two numbers
  })
}

## this is a function to deal with the weird ecotype names
extract_first_number <- function(col) {
  sapply(col, function(x) {
    strsplit(x, "_")[[1]][1] |> as.numeric()
  })
}


Z <- vcfR2tidy(mysnp, format_fields = "GT") #extract the genotype data from the vcf
genoDf <- data.frame(Z$gt) # turn into a data frame
genoDf$genotype = sapply(genoDf$gt_GT, sum_genotype) #turn to a numeric genotype
genoDf$FID = sapply(genoDf$Indiv, extract_first_number) #fix the genotype id

 
traits <- read.table('data/phenotypes.pheno', header=T) #read in trait data
mergedData <- dplyr::inner_join(genoDf, traits, by="FID") # merge in the phenotypes

library(ggplot2)
ggplot(mergedData) + aes(x=as.factor(genotype), y=DTF1)+
  geom_violin()+
geom_jitter()+
  theme_classic()


mymodel = lm(DTF1~genotype, data=mergedData)
summary(mymodel)


```

Compare to a random SNP [insert vcf info here]